comp/core/workloadmeta/collectors/internal/docker/docker.go

```diff
@@ -177,7 +177,11 @@ func (c *collector) stream(ctx context.Context) {
 }
 
 func (c *collector) generateEventsFromContainerList(ctx context.Context, filter *containers.Filter) error {
-	containers, err := c.dockerUtil.RawContainerListWithFilter(ctx, container.ListOptions{}, filter)
+	if c.store == nil {
```

Comment:
maybe check if `c` is not nil ? normally it can't but for security to avoid accessing a field of a nil var ?

Reply:
In Go, it is not common to test if the receiver (`c` in this case) is nil.

Reply:
okay old c habit ^^, mybad